nexus/src/app/utilization.rs

```diff
@@ -50,38 +51,17 @@ impl super::Nexus {
             .db_datastore
             .ip_pool_total_capacity(opctx, &authz_pool)
             .await?;
-
-        // we have one query for v4 and v6 allocated and one for both v4 and
-        // v6 capacity so we can do two queries instead 4, but in the response
-        // we want them paired by IP version
-        Ok(IpPoolUtilization {
-            ipv4: Ipv4Utilization {
-                // This one less trivial to justify than the u128 conversion
-                // below because an i64 could obviously be too big for u32.
-                // In addition to the fact that it is unlikely for anyone to
-                // allocate 4 billion IPs, we rely on the fact that there can
-                // only be 2^32 IPv4 addresses, period.
-                allocated: u32::try_from(allocated.ipv4).map_err(|_e| {
-                    Error::internal_error(&format!(
-                        "Failed to convert i64 {} IPv4 address count to u32",
-                        allocated.ipv4
-                    ))
-                })?,
-                capacity: capacity.ipv4,
-            },
-            ipv6: Ipv6Utilization {
-                // SQL represents counts as signed integers for historical
-                // or compatibility reasons even though they can't really be
-                // negative, and Diesel follows that. We assume here that it
-                // will be a positive i64.
-                allocated: u128::try_from(allocated.ipv6).map_err(|_e| {
-                    Error::internal_error(&format!(
-                        "Failed to convert i64 {} IPv6 address count to u128",
-                        allocated.ipv6
-                    ))
-                })?,
-                capacity: capacity.ipv6,
-            },
-        })
+        let capacity = capacity as f64;
+        let remaining = capacity - allocated as f64;
+        if remaining < 0.0 {
+            return Err(Error::internal_error(
+                format!(
+                    "Computed an impossible negative count of remaining IP \
+                addresses. Capacity = {capacity}, allocated = {allocated}"
+                )
+                .as_str(),
+            ));
+        }
+        Ok(IpPoolUtilization { remaining, capacity })
```

Comment:
Would it make slightly more sense to do the math as ints and then convert to float here at the end?